### PR TITLE
Kp/update pse 7

### PR DESCRIPTION
**Pull request by kyra-at-ada**
The current PSE test cases are insufficient. In fact, the solution presented in the instructor-only section of this PSE is flawed, and the test cases do not catch it. Here's a test that causes the given solution to fail:
```
print(reshape_matrix(
    [[7, 2, 1],
     [4, 3, 5],
     [6, 9, 8]], 9, 1))
```

**Summary of Changes**
- Updated the instructor solution
- Added test case above
- Updated the sample solution

**For review**
- Please preview in learn and ensure sample solution works
- Check for any missing edge cases

### CODE REVIEW

File: 07/reshape-a-matrix.checkpoint.md [hunk 1]
```diff
@@ -109,14 +109,25 @@ def test_cannot_reshape():
...
+def test_convert_4_2_to_2_4():
     matrix = [[1,2],[3,4],[5,6],[7,8]]
     r = 2
     c = 4
 
     reshaped_matrix = reshape_matrix(matrix, r, c)
 
     assert reshaped_matrix == [[1,2,3,4],[5,6,7,8]]
+
+def test_three_by_three_to_nine_by_one(self):
```
Comment (chimerror) [inline 07/reshape-a-matrix.checkpoint.md:121]: Style nitpick: I personally like using the number words here over the numerals, but it might be good to be consistent with the other tests (which I recommend doing by updating the other tests to match this one).

File: 07/reshape-a-matrix.instructor.md [hunk 2]
```diff
@@ -13,18 +13,25 @@ def reshape_matrix(matrix, r, c):
...
-    
-    new_row = []
-    new_matrix = []
-    for i in range(originalRows):
-        for j in range(originalColumns):
-            new_row.append(matrix[i][j])
 
-            if len(new_row) == c:
-                new_matrix.append(new_row)
-                new_row = []
```
Comment (chimerror) [inline 07/reshape-a-matrix.instructor.md:27]: As mentioned in Slack, this solution actually works with the new test case, and was updated by Kelsey in [this commit](https://github.com/Ada-Developers-Academy/core-pses/commit/efb15eceb2ddfd16f1995077387b4cbcb2f6a4fd). However, the solution in the checkpoint _was_ still incorrect but you fixed it.

I'm not too particular that we need to keep both solutions, but I'm pointing out for visibility!

**Review by chimerror — APPROVED** [review]
Made a comment about a very minor style nitpick, as well as the fact that this ticket had already been partially completed before you got to it.

Neither are particularly blocking to me, so I'm going to approve. I did preview your changes and made sure the old broken instructor implementation did indeed fail the new test.
